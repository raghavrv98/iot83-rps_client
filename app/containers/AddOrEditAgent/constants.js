/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
