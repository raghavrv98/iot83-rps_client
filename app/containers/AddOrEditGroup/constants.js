/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
