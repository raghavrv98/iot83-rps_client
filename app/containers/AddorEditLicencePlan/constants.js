/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";