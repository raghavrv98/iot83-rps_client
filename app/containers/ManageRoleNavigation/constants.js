/*
 *
 * ManageRoleNavigation constants
 *
 */

export const DEFAULT_ACTION = "app/ManageRoleNavigation/DEFAULT_ACTION";

export const GET_MENUS = "app/ManageRoleNavigation/GET_MENUS";
export const GET_MENUS_SUCCESS = "app/ManageRoleNavigation/GET_MENUS_SUCCESS";
export const GET_MENUS_FAILURE = "app/ManageRoleNavigation/GET_MENUS_FAILURE";

export const GET_ALL_MENUS = "app/ManageRoleNavigation/GET_ALL_MENUS";
export const GET_ALL_MENUS_SUCCESS = "app/ManageRoleNavigation/GET_ALL_MENUS_SUCCESS";
export const GET_ALL_MENUS_FAILURE = "app/ManageRoleNavigation/GET_ALL_MENUS_FAILURE";

export const SAVE_MENUS = "app/ManageRoleNavigation/SAVE_MENUS";
export const SAVE_MENUS_SUCCESS = "app/ManageRoleNavigation/SAVE_MENUS_SUCCESS";
export const SAVE_MENUS_FAILURE = "app/ManageRoleNavigation/SAVE_MENUS_FAILURE";
