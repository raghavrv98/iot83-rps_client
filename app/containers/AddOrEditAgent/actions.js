/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
