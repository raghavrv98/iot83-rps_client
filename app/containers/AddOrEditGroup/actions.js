/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
