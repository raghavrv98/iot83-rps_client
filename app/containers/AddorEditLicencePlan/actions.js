/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}