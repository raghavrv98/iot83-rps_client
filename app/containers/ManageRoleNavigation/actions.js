import { DEFAULT_ACTION, GET_MENUS, GET_ALL_MENUS, SAVE_MENUS } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function getMenus(id) {
  return {
    type: GET_MENUS,
    id
  };
}

export function getAllMenus() {
  return {
    type: GET_ALL_MENUS,
  };
}

export function saveMenu(payload,id) {
  return {
    type: SAVE_MENUS,
    payload,
    id
  };
}
