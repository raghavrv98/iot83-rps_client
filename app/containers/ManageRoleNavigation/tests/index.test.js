import React, { Component } from 'react';
import { shallow } from 'enzyme';
import history from "utils/history";
import { getMenus, getAllMenus, saveMenu } from "../actions";
import { GET_MENUS, GET_ALL_MENUS, SAVE_MENUS } from "../constants";
import { ManageRoleNavigation, mapDispatchToProps } from "../index";

const list = [
    {
      _id: "9bed0d5ca79011e9863a0242ac110016",
      createdAt: 1563257514466,
      name: "edfedeefdf",
      other: "fvcerdfedfv",
      overallGeometry: "dfe3dfedc",
      plantId: "f29b79c09f0911e986810242ac110011",
      processFluid: "efe3dcfed",
      updatedAt: 1563257547048
    }
  ];

describe('< ManageRoleNavigation />', () => {

    const match = {
        params: {
            roleId: ""
        }
    }

    it('Expect to have unit tests specified', () => {
    expect(true).toEqual(true);
    });

    it('should test component when render', () => {
        spyOn(ManageRoleNavigation.prototype, 'render').and.callThrough();
        const wrapper = shallow(<ManageRoleNavigation getMenus={getMenus} match={match} />);
        expect(wrapper).toBeDefined();
        expect(ManageRoleNavigation.prototype.render).toHaveBeenCalledTimes(1);
    })

    it('match snapshots', () => {
        const wrap = shallow(<ManageRoleNavigation match={match} getMenus={getMenus} />)
        expect(wrap).toMatchSnapshot()
    })

    it("test componentWillMount", async () => {
        const match = {
            params: {
                roleId: "demo"
            }
        };
        const fetchList = jest.fn();
        spyOn(ManageRoleNavigation.prototype, "componentWillMount").and.callThrough();
        shallow(
            <ManageRoleNavigation
                match={match}
                getMenus={getMenus}
            />
        );
        expect(ManageRoleNavigation.prototype.componentWillMount).toHaveBeenCalledTimes(1);
        expect(fetchList).toHaveBeenCalledTimes(0);
    });

    it('should test component when recieve props', () => {
        spyOn(ManageRoleNavigation.prototype, 'componentWillReceiveProps').and.callThrough();
        const wrapper = shallow(<ManageRoleNavigation getMenus={getMenus} match={match} list={undefined} listError={undefined} deleteSuccess={undefined} deleteFailure={undefined} />);
        expect(wrapper).toBeDefined();
        wrapper.setState({ list });
        wrapper.setProps({ list, listError: "No Error Found.", deleteSuccess: "bb157d6738bb4886b77fd25635301010", deleteFailure: "No Error Found in delete." });
        expect(ManageRoleNavigation.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
    })

    it("Function testing on mapDispatchToProps", () => {
        const mockParams = {
          params: {
            id: "demo",
            payload: "demo"
          }
        };
        const dispatch = jest.fn();
        mapDispatchToProps(dispatch).getMenus(
          mockParams.params.id
        );
        expect(dispatch.mock.calls[0][0]).toEqual({
          type: GET_MENUS,
          id: "demo"
        });
        mapDispatchToProps(dispatch).getAllMenus();
        expect(dispatch.mock.calls[1][0]).toEqual({
          type: GET_ALL_MENUS
        });
        mapDispatchToProps(dispatch).saveMenu(
            mockParams.params.id,mockParams.params.payload
          );
          expect(dispatch.mock.calls[2][0]).toEqual({
            type: SAVE_MENUS,
            id: "demo",
            payload: "demo"
        });
    });

});