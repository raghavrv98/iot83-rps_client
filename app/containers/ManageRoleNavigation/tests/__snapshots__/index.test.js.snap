// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< ManageRoleNavigation /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      ManageRoleNavigation
    </title>
    <meta
      content="Description of ManageRoleNavigation"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p
        className="pd-l-30"
      >
        <span
          className="cursor-pointer"
          onClick={[Function]}
        >
          <button
            className="btn btn-transparent"
          >
            <i
              className="far fa-long-arrow-left"
            />
          </button>
          <FormattedMessage
            defaultMessage="Manage Roles"
            id="app.common.messages.manageRoles"
            values={Object {}}
          >
            <Component />
          </FormattedMessage>
        </span>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="Manage Role Navigation"
          id="app.containers.ManageRoleNavigation.manageRoleNavigation"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </h5>
    </div>
    <div
      className="flex-item fx-b30 text-right align-items-center"
    />
  </div>
  <SkeletonLoader
    mode="fullView"
    skeleton="skeletonManageNavigation"
  />
</div>
`;
